keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
-+ COSI unbinds the BR from the `Bucket`.
-+ Sidecar sees the `Bucket` is unbound, and if the retention policy is "Delete", the gRPC calls the provisoner's _Delete_ interface. Upon successful completion, the `Bucket` is updated to Deleted.
-+ If the retention policy is "Retain" then the `Bucket` remains "Released" and it can potentially be reused.
-+ When COSI sees the `Bucket` is "Deleted", it removes all finalizers and the resources are garbage collected.
+The COSI sidecar, which runs alongside each of the drivers, listens for BucketAccess events. All but the sidecar with the specified provisioner will ignore the BucketAccess object. Only the appropriate sidecar will make a gRPC request to its driver to generate credentials for a Bucket in the OSP.
 
-## Grant Bucket Access
-This workflow describes the automation supporting granting access to an existing backend bucket.
+More information about COSI gRPC API is [here](#cosi-grpc-api)
 
-Here is the workflow:
+```
+    COSI Driver (s3.amazonaws.com)
+    |------------------------------------------|
+    | grpc ProvisionerGrantBucketAccess({      |
+    |      "name": "bar-$uuid",                |
+    |      "bucketID": "br-$uuid",             |
+    |      "parameters": {                     |
+    |         "key": "value"                   |
+    |      }                                   |
+    | })                                       |
+    |------------------------------------------|
+```
+Each BucketAccess is meant to map to one account (or its equivalent) in the OSP, and grant access for that account to access the bucket. Once the credentials are successfully generated in the OSP, the credential values are stored in a secret. The secret resides in the namespace of the driver. Since each driver is expected to run in its own namespace, the secret containing access credentials cannot be accessed directly by users.
 
-+ The admin creates a cluster-scoped `Bucket` representing the backend bucket.
-+ The user creates a `BucketRequest` referencing this `Bucket` instance and COSI follows the "Create Bucket" workflow.
-+ The user creates a `BucketAccessRequest` (BAR) which references their BR.
-+ COSI central controller sees a new `BucketAccessRequest` (BAR) and gets the `BAR.BucketAccessClass` (BAC).
-+ COSI creates the cluster-scoped `BucketAccess` (BA) filling in the field from the BAR and BAC.
-+ The sidecar sees the new BA and gRPC calls the driver to grant access to the backend bucket.
-+ COSI updates the BAR and BA to reference each other.
-+ The cosi-node-adapter, which has been waiting for the BAR to be processed, writes access tokens to the app pod's specified mount point.
+The sidecar sets `accessGranted=true` in the BucketAccess object and sets the `credentialsSecret` field to the name and namespace of the generated secret. At this point, the BucketAccess is ready to be utilized by workloads.
```

Comment:
BA.status.AccessGranted (uppercase A) is the name in this resource defined below.

Reply:
yeah exactly. I'm using the yaml representation whenever I specify a single field.